libs/langchain-community/src/chat_models/ibm.ts

```diff
@@ -0,0 +1,859 @@
+    };
+  } else {
+    additional_kwargs = {};
+  }
+
+  if (role === "user") {
+    return new HumanMessageChunk({ content });
+  } else if (role === "assistant") {
+    const toolCallChunks: ToolCallChunk[] = [];
+    if (rawToolCalls && rawToolCalls.length > 0)
+      for (const rawToolCallChunk of rawToolCalls) {
+        toolCallChunks.push({
+          name: rawToolCallChunk.function?.name,
+          args: rawToolCallChunk.function?.arguments,
+          id: rawToolCallChunk.id,
+          index: rawToolCallChunk.index,
+          type: "tool_call_chunk",
+        });
+      }
+
+    return new AIMessageChunk({
+      content,
+      tool_call_chunks: toolCallChunks,
+      additional_kwargs,
+      usage_metadata: {
+        input_tokens: usage?.prompt_tokens ?? 0,
+        output_tokens: usage?.completion_tokens ?? 0,
+        total_tokens: usage?.total_tokens ?? 0,
+      },
+      id: rawData.id,
+    });
+  } else if (role === "tool") {
+    if (rawToolCalls)
+      return new ToolMessageChunk({
+        content,
+        additional_kwargs,
+        tool_call_id: _convertToValidToolId(modelId, rawToolCalls?.[0].id),
+      });
+  } else if (role === "function") {
+    return new FunctionMessageChunk({
+      content,
+      additional_kwargs,
+    });
+  } else {
+    return new ChatMessageChunk({ content, role });
+  }
+  return null;
+}
+
+export class ChatWatsonx<
+    CallOptions extends WatsonxCallOptionsChat = WatsonxCallOptionsChat
+  >
+  extends BaseChatModel<CallOptions>
+  implements ChatWatsonxInput
+{
+  static lc_name() {
+    return "ChatWatsonx";
+  }
+
+  lc_serializable = true;
+
+  get lc_secrets(): { [key: string]: string } {
+    return {
+      authenticator: "AUTHENTICATOR",
+      apiKey: "WATSONX_AI_APIKEY",
+      apikey: "WATSONX_AI_APIKEY",
+      watsonxAIAuthType: "WATSONX_AI_AUTH_TYPE",
+      watsonxAIApikey: "WATSONX_AI_APIKEY",
+      watsonxAIBearerToken: "WATSONX_AI_BEARER_TOKEN",
+      watsonxAIUsername: "WATSONX_AI_USERNAME",
+      watsonxAIPassword: "WATSONX_AI_PASSWORD",
+      watsonxAIUrl: "WATSONX_AI_URL",
+    };
+  }
+
+  get lc_aliases(): { [key: string]: string } {
+    return {
+      authenticator: "authenticator",
+      apikey: "watsonx_ai_apikey",
+      apiKey: "watsonx_ai_apikey",
+      watsonxAIAuthType: "watsonx_ai_auth_type",
+      watsonxAIApikey: "watsonx_ai_apikey",
+      watsonxAIBearerToken: "watsonx_ai_bearer_token",
+      watsonxAIUsername: "watsonx_ai_username",
+      watsonxAIPassword: "watsonx_ai_password",
+      watsonxAIUrl: "watsonx_ai_url",
+    };
+  }
+
+  getLsParams(options: this["ParsedCallOptions"]): LangSmithParams {
+    const params = this.invocationParams(options);
+    return {
+      ls_provider: "watsonx",
+      ls_model_name: this.model,
+      ls_model_type: "chat",
+      ls_temperature: params.temperature ?? undefined,
+      ls_max_tokens: params.maxTokens ?? undefined,
+    };
+  }
+
+  model = "mistralai/mistral-large";
```

Comment:
Generally we suggest not using a default (models change frequently), but up to you